src/components/progress-circle/progress-circle.ts

```diff
@@ -41,47 +84,172 @@ export class MdProgressCircle {
    * mode is bound to the host as the attribute host.
    */
   @HostBinding('attr.mode')
-  @Input() mode: 'determinate' | 'indeterminate' = 'determinate';
+  @Input()
+  get mode() {
+    return this._mode;
+  }
+  set mode(m: 'determinate' | 'indeterminate') {
+    if (m == 'indeterminate') {
+      this._startIndeterminateAnimation();
+    } else {
+      this._cleanupIndeterminateAnimation();
+    }
+    this._mode = m;
+  }
+  private _mode: 'determinate' | 'indeterminate' = 'determinate';
+
+
+  constructor(ref: ChangeDetectorRef) {
+    this._ref = ref;
+  }
 
 
   /**
-   * Gets the current stroke dash offset to represent the progress circle.
-   *
-   * The stroke dash offset specifies the distance between dashes in the circle's stroke.
-   * Setting the offset to a percentage of the total circumference of the circle, fills this
-   * percentage of the overall circumference of the circle.
+   * Animates the rendered circle.
    */
-  strokeDashOffset() {
-    // To determine how far the offset should be, we multiple the current percentage by the
-    // total circumference.
+  private _renderCircle(animateFrom: number, animateTo: number, ease: Function,
```

Comment:
Definitely need a description that explains how this functions works at a high level.